compiler/il/OMRBlock.cpp

```diff
@@ -805,6 +807,687 @@ OMR::Block::splitWithGivenMethodSymbol(TR::ResolvedMethodSymbol *methodSymbol, T
+                     // we check the regStore information pre split point. As regStore post split point is most likely to be used later. This will reduce number of shuffles.
+                     // So if we get the candidate from before split point, we do not need to check list of RegStores post split point.
+                     needToCheckStoreRegPostSplitPoint = false;
+                     }
+                  }
+               }
+   
+            
+            if (nodeInfoEntry->second.second != NULL && nodeInfoEntry->second.second->getOpCode().isLoadReg() && 
+               dep->getLowGlobalRegisterNumber() == nodeInfoEntry->second.second->getLowGlobalRegisterNumber() && 
+               dep->getHighGlobalRegisterNumber() == nodeInfoEntry->second.second->getHighGlobalRegisterNumber())
+               {
+               // Passthrough uses same register as the replacement, do not need to do anything
+               }
+            else if (needToCreateRegStore && (!needToCheckStoreRegPostSplitPoint || !checkStoreRegNodeListForNode(dep, storeRegNodePostSplitPoint)))
+               {
+               // We have either used different register for replacement to value node after split point or Passthrough has corresponding regStore before spilt point, but registers were unavailable.
+               // In these cases we need to create a regStore for the PassThrough.
+               // If we have replacement then need to store the replacement to register, else, value needs to be stored. 
+               TR::Node *nodeToBeStored = nodeInfoEntry->second.second != NULL ? nodeInfoEntry->second.second : value;
+               TR::Node *regStore = TR::Node::create(value, comp->il.opCodeForRegisterStore(value->getDataType()), 1, nodeToBeStored);
+               currentTT->insertBefore(TR::TreeTop::create(comp, regStore));
+               regStore->setGlobalRegisterNumber(dep->getGlobalRegisterNumber());
+               if (nodeToBeStored->requiresRegisterPair(comp))
+                  regStore->setHighGlobalRegisterNumber(dep->getHighGlobalRegisterNumber());
+               TR::SymbolReference *ref = NULL;
+               if (nodeToBeStored->getOpCode().isLoadConst() || nodeToBeStored->getOpCodeValue() == TR::loadaddr || nodeInfoEntry->second.second == NULL)
+                  {
+                  // See if we have stored constants in register before split point.
+                  auto storeRegNodeInfoEntry = storeNodeInfo->find(value);
+                  if (storeRegNodeInfoEntry != storeNodeInfo->end())
+                     {
+                     ref = storeRegNodeInfoEntry->second->getRegLoadStoreSymbolReference();
+                     }
+                  else 
+                     {
+                     TR_ASSERT_FATAL(false, "We have a node under PassThrough and we did not find a regStore that is using the info.");
```

Comment:
How about:

```
                  // See if we have stored constants in register before split point.
                  auto storeRegNodeInfoEntry = storeNodeInfo->find(value);
                  TR_ASSERT_FATAL(storeRegNodeInfoEntry == storeNodeInfo->end(), "We have a node under PassThrough and we did not find a regStore that is using the info.");

                  ref = storeRegNodeInfoEntry->second->getRegLoadStoreSymbolReference();
```

Also is there any other useful information you could put on the assert such that while encountering it in the wild you could use the information to determine the cause?

Reply:
Thanks a lot for the suggestion. I have made the change to add more useful information with this refactoring in https://github.com/eclipse/omr/commit/e75062fc5ab9ebb3996cc60947cb12fb2f027537